docs/index.md

```diff
@@ -27,53 +27,130 @@ can be used to:
 
 ## Instructions
 
-1.  Set up Eclipse and CLDR, according to instructions:
+### General Setup for Maven
+
+1.  Configure Maven settings (including Github tokens) according to these instructions:
     <http://cldr.unicode.org/development/maven>
+2.  Get your github account authorized for <https://github.com/unicode-org/unicodetools>,
+    create a fork under your account, and create a local clone.
+
+#### In-source / Out-of-source build setup
+
+Some of the tasks within the Unicode Tools generate output files that can also be input files into other steps.
+For this purpose, we create a folder named `Generated` to store these files.
+This folder can be a subfolder inside the local working copy root (called an "In-source build" workspace layout), or this folder can be outside (ex: a sibling folder) the local working-copy root (called an "Out-of-source build" workspace layout). Both workspace styles are described below.
+
+Out-of-source builds keep a separation between source files of the repository and their generated output files, which are no tracked in the repository. Out-of-source builds allow developers to maintain a clean view of changes to tracked source files, without mixing generated output files. (Out-of-source builds are also useful for C++ repositories in which multiple configurations can be invoked to generate independent sets of makefiles that result in corresponding different output compiled binary files.
+
+##### Setup for an out-of-source build workspace
+
+1. Create directories for cloning the Unicode Tools and CLDR repositories:
+```
+mkdir -p unicodetools/mine/src
+mkdir -p cldr/mine/src
+```
+2. Clone the repositories' contents into their respective source directories:
+```
+git clone https://github.com/unicode-org/unicodetools.git unicodetools/mine/src
+git clone https://github.com/unicode-org/cldr.git cldr/mine/src
+```
+3. Create the `Generated` folder structure as a sibling to the local working copy root:
+```
+mkdir -p unicodetools/mine/output/Generated/BIN
```

Comment:
The extra "output" folder level seems redundant with "Generated". I suggest removing the "output" layer.

Reply:
The [code expects](https://github.com/unicode-org/unicodetools/blob/main/unicodetools/src/main/java/org/unicode/text/utility/Settings.java#L151-L154) `Generated` to be a subdirectory of the "root of where we write output files". There is [at least one instance](https://github.com/unicode-org/unicodetools/blob/main/unicodetools/src/main/java/org/unicode/tools/Ids.java#L1267) where a different subdirectory within this root output folder is used. Also, getting rid of the output directory structure would make the UNICODETOOLS_OUTPUT_DIR system property value effectively be the parent of the UNICODETOOLS_REPO_DIR.

Do you want to document the meaning of the Java system properties, either here in this PR (ex: adding a 3rd column to the table below) or in the future?

Reply:
> The [code expects](https://github.com/unicode-org/unicodetools/blob/main/unicodetools/src/main/java/org/unicode/text/utility/Settings.java#L151-L154) `Generated` to be a subdirectory of the "root of where we write output files". There is [at least one instance](https://github.com/unicode-org/unicodetools/blob/main/unicodetools/src/main/java/org/unicode/tools/Ids.java#L1267) where a different subdirectory within this root output folder is used.

Yes, I rewrote Settings.java earlier this year to look like this. I intended, documented, and used this with Generated as a sibling of the repo root (my "src"), which means that the UNICODETOOLS_OUTPUT_DIR is the parent folder of where "src" is.

> Also, getting rid of the output directory structure would make the UNICODETOOLS_OUTPUT_DIR system property value effectively be the parent of the UNICODETOOLS_REPO_DIR.

Yes, exactly. As intended.
I am not sure if anything writes anywhere outside of the Generated child of the UNICODETOOLS_OUTPUT_DIR. Either everything or almost everything works inside GEN_DIR.
We would only get a problem if we wrote into a folder directly in UNICODETOOLS_OUTPUT_DIR whose name happened to collide with that of the repo root folder ("src" in my case).

> Do you want to document the meaning of the Java system properties, either here in this PR (ex: adding a 3rd column to the table below) or in the future?

Too much detail. We have comments in Settings.java for that, and sometimes it just boils down to look at which code uses which path.

Reply:
PS: There is another way of collapsing "output" and "Generated": We could change GEN_DIR to be the same as UNICODETOOLS_OUTPUT_DIR rather than a child folder. Then UNICODETOOLS_OUTPUT_DIR could be a sibling to "src" without concerns about a child folder accidentally colliding with the repo root. We would lose the special name "Generated" but that's probably ok; we would need to change other docs pages for that.

That would require a code change. I would first document what we have now, and if we agree to collapse the two levels, then make another PR that changes code & docs.